#This repository contains the supplementary files associated with the article Optimized Wastewater Sample Processing for Antimicrobial Resistance Surveillance.
#These files include R scripts, data tables, and results used in the analyses described in the study. Below is a detailed description of each file:

# Supplementary Files

#S1. script_dada2_WDorWF.R: An R script file with the code used for processing sequencing data through the DADA2 pipeline. The script also included all analyses performed with the Phyloseq package.#
#S2. Mito_OTUs_taxonomy.txt: This file contains the taxonomic assignments for the mitochondrial OTUs.
#S3. OTU.fas: This file contains the representative sequences for each OTU.
#S4. Mito_OTUs_counts.txt: This file contains the OTU abundance table for the mitochondrial OTUs.
#S5. 16S_ASVs_taxonomy.txt: This file contains the taxonomic assignments for the 16S rRNA gene ASVs.
#S6. 16S_ASVs_counts.txt: This file contains the ASV abundance table for the 16S rRNA gene.
#S7. 16S_ASVs.fa: This file contains the representative sequences for each 16S rRNA gene ASV.
#S8. Summary.csv: This file contains summary statistics for DADA2 analysis (all samples included).
#S9. Metadata.xlsx: An Excel file providing the metadata associated with the samples and experiments conducted in the study.
#S10. Pretest_SequenceSummary.xls: An Excel file summarizing statistics for DADA2 analysis only for samples included in preliminary test experiments, as well as complete alpha diversity analysis metrics. 
#S11. Dataset_FilterPellet.xlsx: An Excel workbook containing the compiled data comparing the results from filter and pellet sample processing methods regarding the relative abundance of antimicrobial resistance genes (ARGs).
#S12. Script_FilterPellet_ARGs.R: An R script file containing the code utilized for the statistical analysis and visualization of the data related to the filter and pellet processing comparison within the study.
#S13. Dataset_Kit_ARGs.xlsx: This Excel file presents the data collected from the use of different DNA isolation kits in the study pertaining to their effectiveness in recovering ARGs from wastewater samples.
#S14. Script_Kit_ARGs.R: The R programming language script associated with the analysis of dARGs obtained from the different DNA isolation kits.
#S15. Dataset_Speed_ARGs.xlsx: In this Excel spreadsheet, the effects of varying centrifugation speeds on the recovery of ARGs are documented.
#S16. Script_Speed_ARGs.R: This R script provides the framework for processing and analyzing the data on the impact of different centrifugation speeds on ARG prevalence.
#S17. NanodropQubitDataSet.xlsx: A dataset that includes the DNA quantification results obtained from both the NanoDrop One™ and Invitrogen™ Qubit™ 4 instruments, as well as from qPCR measurements of the 16S rRNA gene.
#S18. Script_NanodropQubit.R: This R script file contains the code used for analyzing and comparing the DNA quantification results from the NanoDrop One™ and Invitrogen™ Qubit™ 4 instruments.
FASTQ files can be found here: https://udemontreal-my.sharepoint.com/:u:/g/personal/helena_ferreira_leal_de_carvalho_toledo_umontreal_ca/EcB_Iw0xEptEgZbFDBEhIo8B_gWp5tpP5vpU2qg-1aAFUA?e=g2V0lA
#Contact
#For inquiries or further information, please contact:
#Helena Ferreira Leal,  MSc | PhD Candidate in Microbiology 
#Département de microbiologie, infectiologie et immunologie - Université de Montréal
#Tél. : +1 (438) 527-8631 
#helena.ferreira.leal.de.carvalho.toledo@umontreal.ca    
